src/test/java/io/harness/cf/client/api/EvaluatorTest.java

```diff
@@ -1,65 +1,291 @@
 package io.harness.cf.client.api;
 
+import static org.junit.Assert.*;
 
+import com.github.benmanes.caffeine.cache.Cache;
 import com.github.benmanes.caffeine.cache.Caffeine;
+import com.google.common.collect.ImmutableList;
 import io.harness.cf.client.dto.*;
+import io.harness.cf.client.dto.Target;
+import io.harness.cf.model.*;
+import java.util.Arrays;
+import java.util.List;
 import org.junit.BeforeClass;
 import org.junit.Test;
 
+/** EvaluatorTest tests the flag configuration rules are correctly evaluated */
 public class EvaluatorTest {
 
   private static Evaluator evaluator;
+  private static Cache<String, Segment> cache;
 
   @BeforeClass
   public static void beforeClass() {
+    cache = Caffeine.newBuilder().maximumSize(10000).build();
+    evaluator = new Evaluator(cache);
```

Comment:
To have a better and more abstract code we need to depend here on an interface that has two possibly three methods such as getFlag, getSegment and we don't care where the data comes from

Reply:
Sure that would be good to add.  Out of the scope for this unit test, but as we try to make the implementations more consistent something we should try and add.